0016/endpoint-time-series-event.avsc

```diff
@@ -0,0 +1,102 @@
+{
+    "namespace":"org.kaaproject.ipc.collector2ts.gen.v1",
+    "name":"TimeSeriesEvent",
+    "type":"record",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"appVersionName",
+            "type":"string",
+            "doc":"Application version name the data is sent for"
+        },
+        {
+            "name":"endpointId",
+            "type":"string",
+            "doc":"Identifier of the endpoint that data points belong to"
+        },
+        {
+            "name":"timeSeriesName",
+            "type":"string",
+            "doc":"Name of the time series that data points belong to"
+        },
+        {
+            "name":"dataPoints",
+            "doc":"Array of endpoint time series data points",
+            "type":{
+                "type":"array",
+                "items":{
+                    "namespace":"org.kaaproject.ipc.collector2ts.gen.v1",
+                    "name":"DataPoint",
+                    "type":"record",
+                    "fields":[
+                        {
+                            "name":"timestamp",
+                            "type":"long",
+                            "doc":"Data point UNIX timestamp in milliseconds"
+                        },
+                        {
+                            "name":"value",
+                            "type":[
+                                "boolean",
+                                "int",
+                                "long",
+                                "float",
+                                "double",
+                                "string",
+                                "bytes"
+                            ],
+                            "doc":"Data point value, can be any primitive or composite type. In case of a composite type, value should be encoded into string or bytes."
+                        },
+                        {
+                            "name":"tags",
+                            "doc":"Array of endpoint time series data points",
+                            "type":{
+                                "type":"array",
```

Comment:
Use map from tag name to tag value?

Reply:
I recall that some Avro libraries have issues with the `map` type. Moreover, it's very troublesome in case of a schema versions mismatch.

Reply:
There was an error in our Avro templates that made maps unusable—no issues in the library itself.

As for schema mismatch, from [Avro specification](https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution):
> - **if both are arrays:**
> This resolution algorithm is applied recursively to the reader's and writer's array item schemas.
> - **if both are maps:**
> This resolution algorithm is applied recursively to the reader's and writer's value schemas.
```